src/coreclr/jit/forwardsub.cpp

```diff
@@ -673,32 +673,24 @@ bool Compiler::fgForwardSubStatement(Statement* stmt)
     // Eg if fwdSubNode is a multi-reg call, parent node must be GT_ASG and the
     // local being defined must be specially marked up.
     //
-    if (fwdSubNode->IsMultiRegCall())
+    if (varTypeIsStruct(fwdSubNode) && fwdSubNode->IsMultiRegCall())
     {
         GenTree* const parentNode = fsv.GetParentNode();
 
         if (!parentNode->OperIs(GT_ASG))
         {
-            JITDUMP(" multi-reg call, parent not asg\n");
+            JITDUMP(" multi-reg struct call, parent not asg\n");
             return false;
         }
 
         GenTree* const parentNodeLHS = parentNode->gtGetOp1();
 
         if (!parentNodeLHS->OperIs(GT_LCL_VAR))
         {
-            JITDUMP(" multi-reg call, parent not asg(lcl, ...)\n");
+            JITDUMP(" multi-reg struct call, parent not asg(lcl, ...)\n");
             return false;
         }
 
-#if defined(TARGET_X86) || defined(TARGET_ARM)
-        if (fwdSubNode->TypeGet() == TYP_LONG)
-        {
-            JITDUMP(" TYP_LONG fwd sub node, target is x86/arm\n");
-            return false;
-        }
-#endif // defined(TARGET_X86) || defined(TARGET_ARM)
-
```

Comment:
Just a question. Why do we delete this code that skips TYP_LONG forward substitute node on x86 and arm target?

Reply:
It should not be necessary.

The restriction on substituting multi-reg nodes stems from the fact the backend (LSRA and codegen) can only handle a few select forms of uses for them:
1) `STORE_LCL_VAR(multi-reg node)`
2) `RETURN(matching multi-reg call | LCL_VAR)`

For structs, these restrictions are enforced by the frontend (mainly importer). For `LONG`s - by decomposition, which means we don't need to worry about them here.